src/emoji/Emoji.js

```diff
@@ -1,6 +1,34 @@
-import { createIconSet } from 'react-native-vector-icons';
-import emojiMap from './emojiMap';
+import {createIconSet} from 'react-native-vector-icons';
+import emojiMap, { emoticonToEmojiMap, commonEmoticonList, textCodeList, emojiToTextCodeMap } from './emojiMap';
 
 const Icon = createIconSet(emojiMap, 'AppleColorEmoji');
 
 export default Icon;
+
+const getEmojiFromEmoticons = (emoticons: string): string => String.fromCodePoint(parseInt(emoticons, 16));
+
+export const replaceEmoticonToEmoji = (text: string) => {
```

Comment:
I am not convinced your approach is any faster than what you can do with a ready-made function.